src/linker/Linker.Dataflow/MethodBodyScanner.cs

```diff
@@ -1124,16 +1124,14 @@ private static void MarkArrayValuesAsUnknown (ArrayValue arrValue, int curBasicB
 				if (isByRef) {
 					MarkArrayValuesAsUnknown (arr, curBasicBlock);
 				}
-				return;
 			}
-
-			if (arr.IndexValues.TryGetValue (index.Value, out ValueBasicBlockPair arrayIndexValue))
-				currentStack.Push (new StackSlot (isByRef ? new ArrayElementReferenceValue (arrayIndexValue.Value) : arrayIndexValue.Value));
+			// Don't try to track refs to array elements. Set it as unknown, then push unknown to the stack
+			else if (isByRef)
+				arr.IndexValues[index.Value] = new ValueBasicBlockPair (UnknownValue.Instance, curBasicBlock);
```

Comment:
If I read it correctly this does push anything - which should break stack balance I think.